src/giskardpy/plugin_pybullet.py

```diff
@@ -359,6 +359,10 @@ def attach_object(self, req):
         except:
             pass
 
+    def attach_existing_obj_to_robot(self, name, link, pose):
+        self.unsafe_get_world().attach_existing_obj_to_robot(name, link, pose)
+        self.remove_object(name)
```

Comment:
This seems weird, the ```attach_existing_obj_to_robot``` already removes the object. If you are doing it just to get rid of the marker, you should probably split up the ```remove_object``` function.